doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -1262,9 +1344,13 @@ A `subject` contains the following subfields:
 A `notification` object contains the following subfields:
 
 + `attributes`: List of attributes to be included in the notification message. If not specified (or empty), all attributes are included
-  in the notification
+  in the notification. It also defines the order in which attributes will appear in notifications when `attrsAs` "value" is used
+  (see "Notification messages" section).
 + `callback` : URL pointing to the service which will be invoked when a notification is generated. A NGSIv2
   compliant server must support `http` URL schema, other schemas (e.g. schemas for web sockets) could also be supported.
++ `headers`: a key-map of HTTP headers that will be included in the notification messages.
```

Comment:
Yes, it is optionl, along with query (I'll modify text to make it clear).

In order to prevent what you mean, some headers could be "protected" or, in other words, the header decided by Orion will take prefference with any matching header defined by the user. I'll also add a comment about it.